### ISSUE TITLE
Destructor is called when class w/ desctructor was imported somewhere but not injected

### ISSUE THREAD

**Issue by L2jLiga** [L2jLiga/fastify-decorators#786]
```typescript
@Service()
export class MyService {
  @Destructor()
  async nothing() {
    throw new Error('Should not be called!');
  }

  static someStaticMethod() {}
}

@Controller('/')
export class MyController {
  someMethod() {
    const nothing = MyService.someStaticMethod();
  }
}
```

Expected: MyService#nothing never called since it was not injected
Actual: MyService#nothing is called

**Comment by L2jLiga** [comment]
Issue should be fixed in this version

[fastify-decorators-3.14.1.tgz](https://github.com/L2jLiga/fastify-decorators/files/9939326/fastify-decorators-3.14.1.tgz)

**Comment by L2jLiga** [comment]
Published as [v3.15.0](https://github.com/L2jLiga/fastify-decorators/releases/tag/v3.15.0)
